tests/Integration/Behaviour/Features/Context/Domain/CarrierFeatureContext.php

```diff
@@ -0,0 +1,482 @@
+            $this->getShippingMethodValueMap()[$data['shipping_method']],
+            (int) $data['tax_rules_group_id'],
+            $this->getOutOfRangeBehaviorValueMap()[$data['out_of_range_behavior']],
+            $this->formatShippingRanges($data['ranges_from'], $data['ranges_to'], $data['zone_ids'], $data['prices']),
+            (int) $data['max_width'],
+            (int) $data['max_height'],
+            (int) $data['max_depth'],
+            (float) $data['max_weight'],
+            explode(',', $data['group_ids']),
+            explode(',', $data['shop_ids']),
+            $data['module_name'],
+            PrimitiveUtils::castStringBooleanIntoBoolean($data['module_needs_core_shipping_price'])
+        );
+
+        /** @var CarrierId $carrierId */
+        $carrierId = $this->getCommandBus()->handle($command);
+
+        SharedStorage::getStorage()->set($reference, new Carrier($carrierId->getValue()));
+    }
+
+    /**
+     * @Then /^Carrier "(.+)" shipping price should be calculated by (module|PrestaShop)$/
+     */
+    public function assertShippingPriceIsCalculatedBy($reference, $calculationMethod)
+    {
+        /** @var Carrier $carrier */
+        $carrier = SharedStorage::getStorage()->get($reference);
+
+        $isModule = 'module' === $calculationMethod;
+        $actualResult = (bool) $carrier->shipping_external;
+
+        if ($isModule !== $actualResult) {
+            throw new RuntimeException(sprintf(
+                'Carrier "%s" shipping price is calculated by %s, but it was expected it to be calculated by %s',
+                $reference,
+                $actualResult ? 'module' : 'PrestaShop',
+                $calculationMethod
+            ));
+        }
+    }
+
+    /**
+     * @Then /^Carrier "(.+)" module (should|should not) need the shipping price calculated by PrestaShop$/
+     */
+    public function assertModuleNeedsCoreShippingPrice($reference, $condition)
+    {
+        /** @var Carrier $carrier */
+        $carrier = SharedStorage::getStorage()->get($reference);
+        $needsPrice = 'should' === $condition;
+        $actualResult = (bool) $carrier->need_range;
+
+        if ($needsPrice !== $actualResult) {
+            throw new RuntimeException(sprintf(
+                'Carrier "%s" module %s the shipping price calculated by PrestaShop, but was expected it to %s',
+                $reference,
+                $actualResult ? 'needs' : 'does not need',
+                $condition === 'should' ? 'need' : 'not need'
+            ));
+        }
+    }
+
+    /**
+     * @Then /^Carrier "(.+)" (should|should not) belong to module$/
+     */
+    public function assertIsModule($reference, $condition)
+    {
+        $isModule = 'should' === $condition;
+
+        /** @var Carrier $carrier */
+        $carrier = SharedStorage::getStorage()->get($reference);
+        $actualResult = (bool) $carrier->is_module;
```

Comment:
This var seems redundant.

Reply:
The bool cast is necessary and this var is used couple times, so it seems ok to me